pickhardtpayments/SyncSimulatedPaymentSession.py

```diff
@@ -249,6 +249,7 @@ def _attempt_payments(self, payments):
             if success:
                 self._uncertainty_network.allocate_amount_on_path(
                     attempt["path"], attempt["amount"])
+                self._oracle.settle_payment(attempt["path"], attempt["amount"])
```

Comment:
as mentioned in #16 I am pretty sure this is exactly not the position in the code where this should be done. The reason is that from a lighting perspective we only settle payments if all onions have successfully arrived. Thus if the payment loop has no residual amount that needs to be delivered. This ensures atomicity of payments.

Reply:
fix in a131eaf